spring-boot-project/spring-boot-docs/src/docs/asciidoc/using/devtools.adoc

```diff
@@ -217,10 +217,18 @@ For IntelliJ IDEA, you can follow the https://www.jetbrains.com/help/idea/spring
 As described earlier in the <<using#using.devtools.restart.restart-vs-reload>> section, restart functionality is implemented by using two classloaders.
 For most applications, this approach works well.
 However, it can sometimes cause classloading issues.
+These might manifest in an exception like:
+
+[source,text]
+----
+Caused by: java.lang.IllegalArgumentException: non-public interface is not defined by the given loader
+----
 
 By default, any open project in your IDE is loaded with the "`restart`" classloader, and any regular `.jar` file is loaded with the "`base`" classloader.
 If you work on a multi-module project, and not every module is imported into your IDE, you may need to customize things.
-To do so, you can create a `META-INF/spring-devtools.properties` file.
+The same is true if you use `mvn spring-boot:run` or `gradle bootRun` in a multi-module project: The project containing your `@SpringBootApplication` is loaded with the "`restart`" classloader, and everything else with the "`base`" classloader.
+
+To instruct Spring Boot to load `.jar` files with a different classloader, you can create a `META-INF/spring-devtools.properties` file.
```

Comment:
I'm not sure that this is needed. If it is, I think it should be part of the next paragraph rather than in a paragraph of its own.